mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -85,9 +85,8 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
    * Attempts to safely cast a user/item id to an Int. Throws an exception if the value is
    * out of integer range or contains a fractional part.
    */
-  protected[recommendation] val checkedCast = udf { (n: Any) =>
+  protected[recommendation] val checkedCast = udf { n: AnyRef =>
```

Comment:
I'm fine with this workaround, but I think we should create an expression for this `checkedCast`. The current UDF doesn't work well with inputs of different types.

Reply:
This doesn't even actually work, now that I dig further. Using `Number` doesn't work either. There are a number of UDFs that fail now in MLlib unfortunately.